src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -155,6 +155,12 @@ ciMethod::ciMethod(const methodHandle& h_m, ciInstanceKlass* holder) :
     ciReplay::initialize(this);
   }
 #endif
+
+  if (CompilerOracle::should_blackhole(h_m) &&
+      _signature->return_type()->basic_type() == T_VOID &&
+      h_m->intrinsic_id() == vmIntrinsics::_none) {
```

Comment:
Why not switch to blackhole unconditionally and override existing intrinsic if any?

Reply:
I thought existing intrinsic should take precedence over blackhole. But maybe blackhole should really take a precedence here, as it comes from an explicit compile command. I'll drop `== none` check.